reana_db/models.py

```diff
@@ -252,6 +258,18 @@ def has_exceeded_quota(self):
             r.quota_limit != 0 and r.quota_used >= r.quota_limit for r in self.resources
         )
 
+    def get_workflow_overload_priority(self):
+        """Get priority factor based on the number of current workflows ``running``."""
+        max_concurrent_workflows = REANA_MAX_CONCURRENT_BATCH_WORKFLOWS
+        running_count = self.workflows.filter(
+            or_(
+                Workflow.status == RunStatus.pending,
+                Workflow.status == RunStatus.running,
+            )
+        ).count()
+        priority = round(1 - running_count / max_concurrent_workflows, 2)
```

Comment:
If  `running_count` will be bigger then `max_concurrent_workflows` the overload priority will start increasing the workflow priority (in `get_priority()`) instead of decreasing it. That's why I've added additional checks over [here](https://github.com/reanahub/reana-server/blob/master/reana_server/complexity.py#L38)

Reply:
If I understand it correctly, `running_count` can't be greater than `max_concurrent_workflows` due to [this check](https://github.com/reanahub/reana-server/blob/9bddf50060929313925ae82fa5a377697e842504/reana_server/scheduler.py#L68-L75), but I might be wrong.

Reply:
I guess you are right, but in case someone will modify this logic in the future we could easily exceed `MQ_MAX_PRIORITY` and break the scheduler. So I wonder if we should introduce some check inside `get_priority()` just to be extra safe?